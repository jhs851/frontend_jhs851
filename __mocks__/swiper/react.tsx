import React from 'react'

const mock: React.FC = ({ children }) => <div>{children}</div>

export default mock
