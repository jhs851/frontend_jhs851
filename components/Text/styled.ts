import styled, { css } from 'styled-components'
import { Colors } from 'types/theme'

export const StyledText = styled.p<{
  size: number
  weight?: 500 | 600 | 700
  letterSpacing?: number
  color?: keyof Colors
  ellipsis: boolean
}>`
  font-family: AppleSDGothicNeo, sans-serif;
  font-size: ${({ size }) => `${size}px`};
  font-weight: ${({ weight }) => weight || 500};
  letter-spacing: ${({ letterSpacing }) => `${letterSpacing || -0.1}`}px;
  color: ${({ theme, color }) => theme[color || 'foreground']};

  ${({ ellipsis }) =>
    ellipsis &&
    css`
      overflow: hidden;
      text-overflow: ellipsis '..';
      white-space: nowrap;
    `}
`
