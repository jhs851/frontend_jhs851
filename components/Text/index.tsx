import React from 'react'
import { StyledText } from './styled'
import { Colors } from 'types/theme'

type P = {
  size: number
  weight?: 500 | 600 | 700
  letterSpacing?: number
  color?: keyof Colors
  className?: string
  ellipsis?: boolean
}

const Text: React.FC<P> = ({
  size,
  weight,
  letterSpacing,
  color,
  ellipsis = false,
  className,
  children,
}) => (
  <StyledText
    size={size}
    weight={weight}
    letterSpacing={letterSpacing}
    color={color}
    ellipsis={ellipsis}
    className={className}
  >
    {children}
  </StyledText>
)

export default Text
