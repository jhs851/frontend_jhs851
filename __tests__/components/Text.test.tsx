import { render } from '@testing-library/react'
import Text from 'components/Text'

describe('Text 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    const text = 'Hello World'
    const rendered = render(<Text size={14}>{text}</Text>)

    expect(rendered.queryByText(text)).not.toBeNull()
    expect(rendered.container).toMatchSnapshot()
  })
})
