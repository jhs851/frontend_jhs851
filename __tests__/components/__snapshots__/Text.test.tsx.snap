// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text 컴포넌트 렌더링이 올바르게 된다 1`] = `
.c0 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

<div>
  <p
    class="c0"
    size="14"
  >
    Hello World
  </p>
</div>
`;
